src/utils.py

```diff
@@ -336,7 +322,90 @@ def _terminal(self, state: State, memory: Dict, objects: Sequence[Object],
         if current_index < len(self._children) - 1:
             return False
         current_child = self._children[current_index]
-        return current_child.terminal(state, memory, objects, params)
+        child_memory = memory["child_memory"][current_index]
+        return current_child.terminal(state, child_memory, objects, params)
+
+
+class SingletonParameterizedOption(ParameterizedOption):
+    """A parameterized option that takes a single action and stops.
+
+    For convenience:
+        * Initiable defaults to always True.
+        * Types defaults to [].
+        * Params space defaults to Box(0, 1, (0, )).
+    """
+
+    def __init__(
+        self,
+        name: str,
+        policy: Callable[[State, Dict, Sequence[Object], Array], Action],
+        types: Optional[Sequence[Type]] = None,
+        params_space: Optional[Box] = None,
+        initiable: Optional[Callable[[State, Dict, Sequence[Object], Array],
+                                     bool]] = None
+    ) -> None:
+        if types is None:
+            types = []
+        if params_space is None:
+            params_space = Box(0, 1, (0, ))
+        if initiable is None:
+            initiable = lambda _1, _2, _3, _4: True
+
+        # Wrap the given initiable so that we can track whether the action
+        # has been executed yet.
+        def _initiable(state: State, memory: Dict, objects: Sequence[Object],
+                       params: Array) -> bool:
+            if "start_state" in memory:
+                assert state.allclose(memory["start_state"])
+            # Always update the memory dict due to the "is" check in _terminal.
+            memory["start_state"] = state
+            assert initiable is not None
+            return initiable(state, memory, objects, params)
+
+        def _terminal(state: State, memory: Dict, objects: Sequence[Object],
+                      params: Array) -> bool:
+            del objects, params  # unused
+            assert "start_state" in memory, \
+                "Must call initiable() before terminal()."
+            return state is not memory["start_state"]
+
+        super().__init__(name,
+                         types,
+                         params_space,
+                         policy=policy,
+                         initiable=_initiable,
+                         terminal=_terminal)
+
+
+def action_sequence_to_parameterized_option(
```

Comment:
let's remove this if it's unused. it's in the commit history if we want to come back to it